src/test/java/com/example/customerdb/controller/PurchaseInfoControllerTest.java

```diff
@@ -84,67 +85,142 @@ public class PurchaseInfoControllerTest {
-        verify(purchaseInfoService, times(1)).findAll();
-    }
-
-    @Test
-    public void 新規データが登録できること() throws Exception {
-        // Create a mock PurchaseInfo
-        PurchaseInfo mockPurchaseInfo = new PurchaseInfo();
+        doReturn(purchaseInfos).when(purchaseInfoServiceImpl).findAll();
 
-        // Call the controller method
-        purchaseInfoController.addInfo(mockPurchaseInfo);
+        List<PurchaseInfoResponse> result = purchaseInfoController.getPriceList();
 
-        // Verify the service method was called
-        verify(purchaseInfoService, times(1)).addInfo(mockPurchaseInfo);
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/pricelist")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(3)))
+                .andExpect(MockMvcResultMatchers.content().json("""
+                        [
+                            {
+                                "price": 8000
+                            },
+                            {
+                                "price": 4500
+                            },
+                            {
+                                "price": 1980
+                            }
+                        ]
+                        """));
     }
 
     @Test
-    public void PUTで入力されたデータに更新できること() throws Exception {
-        // Create a mock PurchaseInfo
-        PurchaseInfo mockPurchaseInfo = new PurchaseInfo();
-        int id = 1;
-
-        // Call the controller method
-        purchaseInfoController.updateInfo(id, mockPurchaseInfo);
-
-        // Verify the service method was called with the correct arguments
-        verify(purchaseInfoService, times(1)).updateInfo(id, mockPurchaseInfo);
+    public void 新規でデータを登録できること() throws Exception {
+        PurchaseInfo purchaseInfo = new PurchaseInfo(4, "kai", "ddd@example.com", null, 9999);
+        when(purchaseInfoServiceImpl.addInfo(purchaseInfo)).thenReturn(purchaseInfo);
+
+        String requestBody = """
+                {
+                    "name": "ziro",
+                    "email": "ziro@example.com",
+                    "price": 1900
+                }
+                """;
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .post("/purchase-info")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(requestBody))
+                .andExpect(status().isOk())
+                .andExpect(MockMvcResultMatchers.content().json("""
+                        {
+                             "content": {
+                                    "id": 0,
+                                    "name": "ziro",
+                                    "email": "ziro@example.com",
+                                    "purchaseDate": null,
+                                    "price": 1900
+                                }
```

Comment:
こちらレビューすり抜けていましたね。
もし更新後の状態を返すのであれば
```suggestion
                             "content": {
                                    "id": 4,
                                    "name": "ziro",
                                    "email": "ziro@example.com",
                                    "purchaseDate": null,
                                    "price": 1900
                                }
```
のようにidは4を返すのがよいですね。

Reply:
下記ケースも横展開しないといけません。
- PATCHで指定されたidのデータが更新できること
- PUTで指定されたidのデータが更新できること

Reply:
idに関してはDB側で自動採番しているため、JSONでの入力がされない想定です。
この場合に返却されるidの値は何を返すべきでしょうか？

Reply:
変更対象のIDですね。
PATCHの場合、IDを指定して部分更新するのでレスポンスにIDを含めることは可能なはずです！

Reply:
ちなみにControllerのテストではあまり気にする必要はないのですが、MyBatisには登録時に自動で採番されたIDを取得する方法はあります。
mybatis auto incrementで調べると良いです。

この実装は全受講者さん苦労するところではありますけどね。

Reply:
69afe6966ee201e03cb9b5d2053fc0360e60231d
修正しました！